riff-raff/app/utils/Forms.scala

```diff
@@ -1,21 +1,51 @@
 package utils
 
-import play.api.data.{FormError, Mapping}
-import java.util.UUID
+import com.gu.management.Loggable
+import magenta.input.DeploymentId
+import play.api.data.FormError
 import play.api.data.Forms._
-import play.api.data.format.Formatter
 import play.api.data.format.Formats._
+import play.api.data.format.Formatter
+
+object Forms extends Loggable {
+  val deploymentId = of[DeploymentId](new Formatter[DeploymentId] {
+    def bind(key: String, data: Map[String, String]) = {
+      stringFormat.bind(key, data).right.flatMap{ s =>
+        stringToId(s).toRight(Seq(FormError("form.deployment-id", "Couldn't parse as deployment ID")))
+      }
+    }
+    def unbind(key: String, value: DeploymentId) = {
+      val output = Map(key -> idToString(value))
+      logger.info(s"deploymentId/unbind: $output")
+      output
+    }
+  })
 
-object Forms {
-  val uuid: Mapping[UUID] = of[UUID](new Formatter[UUID] {
-    override val format = Some("format.uuid", Nil)
+  val deploymentIdList = of[List[DeploymentId]](new Formatter[List[DeploymentId]] {
     def bind(key: String, data: Map[String, String]) = {
-      stringFormat.bind(key, data).right.flatMap { s =>
-        scala.util.control.Exception.allCatch[UUID]
-          .either(UUID.fromString(s))
-          .left.map(e => Seq(FormError(key, "error.uuid", Nil)))
+      stringFormat.bind(key, data).right.flatMap{ s =>
+        Right(stringToIdList(s))
       }
     }
-    def unbind(key: String, value: UUID) = Map(key -> value.toString)
+    def unbind(key: String, value: List[DeploymentId]) = {
+      val output = Map(key -> idToString(value))
+      logger.info(s"deploymentIdList/unbind: $output")
+      output
+    }
   })
+
+  // serialisation and de-serialisation code for deployment IDs
+  val DEPLOYMENT_DELIMITER = '!'
+  val FIELD_DELIMITER = '*'
+  def idToString(d: List[DeploymentId]): String = {
```

Comment:
I'd rather have this kind of logic in the companion object (possibly as a type class) than away in this Forms file. Seems eminently testable and a nice candidate for property based testing.